libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -143,18 +160,19 @@ class GrayboxPromote extends LitElement {
       //   });
     };
 
-    this.getValuesTask = new Task(this, {
+    this.setupTask = new Task(this, {
       task: async () => {
-        const { ref, repo, owner, referrer } = getAemInfo();
-        const url = new URL(referrer);
-        const { experienceName, grayboxIoEnv } = await getProjectInfo(referrer);
+        const { ref, repo, owner, excelRef } = getAemInfo();
+        const filePath = await getFilePath({ ref, repo, owner, excelRef });
+        const previewUrl = await getPreviewUrl({ owner, repo, ref, filePath });
+        const { experienceName, grayboxIoEnv } = await getProjectInfo(previewUrl);
```

Comment:
[Task is already catching the errors ](https://github.com/adobecom/milo/pull/2555/files#diff-7214629f44f7a1dbc0bdfc18fb6309965da48000fe961e4aaeed56046e9aa06fR219) from what I can tell, so unless there is value in having specific error messages - just having the error bubble up would be enough IMO 🤔

Reply:
Correct, the task is catching the errors [in here](https://github.com/adobecom/milo/pull/2555/files#diff-7214629f44f7a1dbc0bdfc18fb6309965da48000fe961e4aaeed56046e9aa06fR290). I removed some of the unnecessary try/catch, but left some manual throwing of errors for debugging purposes.